lambda/session/lambda_functions.py

```diff
@@ -95,17 +117,23 @@ def list_sessions(event: dict, context: dict) -> List[Dict[str, Any]]:
     logger.info(f"Listing sessions for user {user_id}")
     sessions = _get_all_user_sessions(user_id)
     resp = []
-    for session in sessions:
-        resp.append(
-            {
-                "sessionId": session["sessionId"],
-                "firstHumanMessage": next(
-                    (msg["content"] for msg in session.get("history", []) if msg.get("type") == "human"), ""
-                ),
-                "startTime": session.get("startTime", None),
-                "createTime": session.get("createTime", None),
-            }
-        )
+
+    with ThreadPoolExecutor() as executor:
```

Comment:
FWIW I think it is more common to reuse the same `ThreadPoolExecutor`. So having something like this at the top of the file and then not creating a new one every time. Then you wouldn't have to have these nested methods.

```python
executor = ThreadPoolExecutor(max_workers=10)
```